packages/react-ui/components/SidePage/SidePage.tsx

```diff
@@ -226,14 +226,29 @@ export class SidePage extends React.Component<SidePageProps, SidePageState> {
   private disablePageScroll = (e: WheelEvent) => {
     const layout = this.layout;
     if (!layout) return;
-    const reachedTop = layout.scrollTop <= 0 && e.deltaY < 0;
-    const reachedBottom = layout.scrollTop >= layout.scrollHeight - layout.offsetHeight && e.deltaY > 0;
-
-    if (!this.props.blockBackground && (reachedTop || reachedBottom)) {
+    const scrollingUp = e.deltaY < 0;
+    const scrollContainerScrollAvailable = this.checkScrollContainerScrollAvailability(layout, scrollingUp);
+    const reachedTop = layout.scrollTop <= 0 && scrollingUp;
+    const reachedBottom = layout.scrollTop >= layout.scrollHeight - layout.offsetHeight && !scrollingUp;
+    const scrollAvailable = !reachedTop && !reachedBottom;
+    if (!this.props.blockBackground && !scrollAvailable && !scrollContainerScrollAvailable) {
```

Comment:
Здесь код полностью дублируется из метода `checkScrollContainerScrollAvailability()`.

```suggestion
    const scrollAvailable = this.checkScrollContainerScrollAvailability(layout, scrollingUp);
    if (!this.props.blockBackground && !scrollAvailable) {
```

Reply:
Нет, это разное. Функция scrollContainerScrollAvailable проверяет возможность скролла для скрол контейнера. Строчки 231-233 проверяют возможность скролла для самой sidePage. Например, проверь с таким кодом:

import { Button } from '@skbkontur/react-ui';

const [opened, setOpened] = React.useState(false);

function renderSidePage() {
  return (
    <SidePage onClose={close} blockBackground>
      <SidePage.Header>Title</SidePage.Header>
      <SidePage.Body>
        <div
          style={{
            background: `repeating-linear-gradient(
                          60deg,
                          #fafafa,
                          #fafafa 20px,
                          #dfdede 20px,
                          #dfdede 40px
                        )`,
            height: 6000,
            padding: '20px 0',
          }}
        >
          <SidePage.Container>
            <p>Use rxjs operators with react hooks</p>
          </SidePage.Container>
        </div>
      </SidePage.Body>
      <SidePage.Footer panel>
        <Button onClick={close}>Close</Button>
      </SidePage.Footer>
    </SidePage>
  );
}

function open() {
  setOpened(true);
}

function close() {
  setOpened(false);
}

<div>
  {opened && renderSidePage()}
  <Button onClick={open}>Open</Button>
</div>;

Reply:
Угу. Код всё равно дублируется. Стоит вынести выборку `querySelector` выше.